src/app/features/tech-record/components/tech-record-summary/tech-record-summary.component.ts

```diff
@@ -50,30 +56,48 @@ export class TechRecordSummaryComponent implements OnInit {
   @ViewChild(TyresComponent) tyres!: TyresComponent;
 
   @Input() vehicleTechRecord!: TechRecordModel;
-
-  private _isEditing: boolean = false;
-  get isEditing(): boolean {
-    return this._isEditing;
-  }
+  @Input() refDataState!: ReferenceDataState;
   @Input()
   set isEditing(value: boolean) {
     this._isEditing = value;
     this.toggleReasonForCreation();
     this.calculateVehicleModel();
   }
-
   @Output() formChange = new EventEmitter();
 
-  vehicleTechRecordCalculated!: TechRecordModel;
+  private _isEditing: boolean = false;
 
+  vehicleTechRecordCalculated!: TechRecordModel;
   sectionTemplates: Array<FormNode> = [];
+  dtpNumbersFromRefData: FormNodeOption<string>[] = [];
 
-  constructor(private store: Store<TechnicalRecordServiceState>) {}
+  constructor(
+    private store: Store<TechnicalRecordServiceState>,
+    private optionsService: MultiOptionsService,
+    private referenceDataStore: Store<ReferenceDataState>
+  ) {}
 
-  ngOnInit(): void {
+  async ngOnInit(): Promise<void> {
```

Comment:
Is this actually async?

Reply:
Ah nah, ty, it was an experiment to combat race condition issues...